src/elementary_functions.jl

```diff
@@ -133,45 +139,50 @@ by indsnetwork"""
 function polynomial_itensornetwork(
   s::IndsNetworkMap,
   coeffs::Vector;
-  dim::Int=default_dimension(),
+  dim::Int=default_dim(),
   k=default_k_value(),
   c=default_c_value(),
 )
   n = length(coeffs)
+  n == 1 && return const_itn(s; c=first(coeffs))
+
   coeffs = [c * (k^(i - 1)) for (i, c) in enumerate(coeffs)]
   #First treeify the index network (ignore edges that form loops)
   _s = indsnetwork(s)
   g = underlying_graph(_s)
   g_tree = undirected_graph(random_bfs_tree(g, first(vertices(g))))
   s_tree = add_edges(rem_edges(_s, edges(g)), edges(g_tree))
   s_tree = IndsNetworkMap(s_tree, indexmap(s))
+  eltype = indexmaptype(s) == RealIndsNetworkMap ? Float64 : ComplexF64
```

Comment:
This reminds me, we need to set up the `Adapt` system so we can convert ITN Functions between types

Reply:
Good point. Yes we should sort that out in a near term PR